### PR TITLE
[LUM-429] Synchronize voters and depositors

### PR DESCRIPTION
**Pull request by Ricardo-Remy**
**User Story:**
For each proposal, we should be able to see who deposited and who voted on it.
Since that's not an information we can compute "on-the-fly", we need to synchronize it to a new table "proposal_votes" and "proposal_deposits", then return it on a paginated call in the proposal controller.
Sample available here https://www.mintscan.io/lum/proposals/19

**Acceptance Criteria:**
- [x] sync depositors and voters from proposalId
- [x] save depositors and voters in DB in two separate table - proposals_vote and proposals_deposits
- [x] Updated voters if they change vote during the voting period
- [x] return paginated call for governance/proposals/:id/voters
- [x] return paginated call for governance/proposals/:id/depositors
- [x] we return 5 depositors and 5 voters per page to simulate a Mintscan behavior

**How to test this feature:**

1. Run `docker-compose up`
2. Run `yarn start:api:dev`
4. Run `yarn test:e2e` -> At this point, before running the test, we seed the db's table proposals_votes and proposals_deposits with some seed data. (Avoid to run the scheduler in parallel)
5. If there are no current proposal votes ongoing. Seed the db before testing the endpoints. 
6. Test the endpoints with an API platform like postman, curl (your choice). For depositors use `/governance/proposals/21/depositors` and for voters use '/governance/proposals/21/voters'
7. You can test the pagination on the voters endpoint by trying to add page params '/governance/proposals/21/voters?page=1'

### CODE REVIEW

File: src/http/responses/proposal.response.ts [hunk 1]
```diff
@@ -41,45 +42,97 @@ class ContentResponse {
...
+    @Expose({ name: 'finalTallyResult' })
     @Type(() => ResultResponse)
     final_result: ResultResponse;
 
     constructor(data: Partial<ProposalResponse>) {
         Object.assign(this, data);
     }
 }
+
+export class ProposalVotersResponse {
```
Comment (Segfaultd) [inline src/http/responses/proposal.response.ts:88]: I believe this should have its own file

File: src/http/responses/proposal.response.ts [hunk 2]
```diff
@@ -41,45 +42,97 @@ class ContentResponse {
...
+    @ApiProperty()
+    @Expose({ name: 'voteOption' })
+    vote_weight: string;
+
+    constructor(data: Partial<ProposalVotersResponse>) {
+        Object.assign(this, data);
+    }
+}
+
+export class ProposalDepositorsResponse {
```
Comment (Segfaultd) [inline src/http/responses/proposal.response.ts:118]: I believe this should have its own file

File: src/services/proposals-deposits.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { Repository } from 'typeorm';
+
+import { ProposalsDepositsEntity } from '@app/database';
+import { Coin } from '@lum-network/sdk-javascript/build/codec/cosmos/base/v1beta1/coin';
+
+@Injectable()
+export class ProposalsDepositsService {
```
Comment (Segfaultd) [inline src/services/proposals-deposits.service.ts:9]: Service should be named singular. Please have a look to other services

File: src/services/proposals-deposits.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    constructor(@Inject('PROPOSALS_DEPOSITS_REPOSITORY') private readonly _repository: Repository<ProposalsDepositsEntity>) {}
+
+    getById = async (proposalId: number): Promise<ProposalsDepositsEntity> => {
```
Comment (Segfaultd) [inline src/services/proposals-deposits.service.ts:12]: This should be named `getByProposalID` and we should have a `getById` method that takes a string (composite key)

File: src/services/proposals-votes.service.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+import { ProposalsVotesEntity } from '@app/database';
+import { LumConstants, LumUtils } from '@lum-network/sdk-javascript';
+import { LumBech32PrefixValAddr } from '@lum-network/sdk-javascript/build/constants';
+
+@Injectable()
+export class ProposalsVotesService {
+    constructor(@Inject('PROPOSALS_VOTES_REPOSITORY') private readonly _repository: Repository<ProposalsVotesEntity>) {}
+
+    getById = async (proposalId: number): Promise<ProposalsVotesEntity> => {
```
Comment (Segfaultd) [inline src/services/proposals-votes.service.ts:13]: This should be named `getByProposalID` and we should have a `getById` method that takes a string (composite key)

**Review by Segfaultd — CHANGES_REQUESTED** [review]

File: src/http/controllers/governance.controller.ts [hunk 6]
```diff
@@ -66,7 +58,57 @@ export class GovernanceController {
         }
 
         return {
-            result: plainToInstance(ResultResponse, result.tally)
+            result: plainToInstance(ResultResponse, result.tally),
         };
     }
+
+    @ApiOkResponse({ status: 200, type: ProposalVotersResponse })
+    // Return only 5 voters per page
```
[the remark this answers is not in the document]
Comment (Ricardo-Remy) [reply src/http/controllers/governance.controller.ts:66]: As discussed, this should be part of enforcing comments across a growing codebase.

**Pull request merged**